app/components/common_post_options/save_option.tsx

```diff
@@ -17,25 +17,34 @@ type CopyTextProps = {
     postId: string;
 }
 
+const messages = defineMessages({
+    save: {
+        id: 'mobile.post_info.save',
+        defaultMessage: 'Save',
+    },
+    unsave: {
+        id: 'mobile.post_info.unsave',
+        defaultMessage: 'Unsave',
+    },
+});
+
 const SaveOption = ({bottomSheetId, isSaved, postId}: CopyTextProps) => {
     const serverUrl = useServerUrl();
 
     const onHandlePress = useCallback(async () => {
         const remoteAction = isSaved ? deleteSavedPost : savePostPreference;
         await dismissBottomSheet(bottomSheetId);
         remoteAction(serverUrl, postId);
-    }, [bottomSheetId, postId, serverUrl]);
+    }, [bottomSheetId, isSaved, postId, serverUrl]);
 
-    const id = isSaved ? t('mobile.post_info.unsave') : t('mobile.post_info.save');
-    const defaultMessage = isSaved ? 'Unsave' : 'Save';
+    const message = isSaved ? messages.unsave : messages.save;
 
     return (
         <BaseOption
-            i18nId={id}
-            defaultMessage={defaultMessage}
+            message={message}
             iconName='bookmark-outline'
             onPress={onHandlePress}
-            testID={`post_options.${defaultMessage.toLocaleLowerCase()}_post.option`}
+            testID={`post_options.${message.defaultMessage.toLocaleLowerCase()}_post.option`}
```

Comment:
Seems that the id would be a better choice here

Reply:
Since it is a test id, I would just hard code it, and not rely on the translation string. I was planning on keeping it as is (since it is, in theory, out of the scope) but I can piggyback the change.

Also, using directly the translation string id is not great, since the translation string id is usually longer and not relevant for this test id.